src/julia_threads.h

```diff
@@ -160,14 +160,8 @@ typedef struct {
     arraylist_t *last_remset;
 
     // variables for allocating objects from pools
-#ifdef _P64
-#  define JL_GC_N_POOLS 49
-#elif MAX_ALIGN == 8
-#  define JL_GC_N_POOLS 50
-#else
-#  define JL_GC_N_POOLS 51
-#endif
-    jl_gc_pool_t norm_pools[JL_GC_N_POOLS];
+#define JL_GC_N_MAX_POOLS 51 // conservative. must be kept in sync with `src/julia_internal.h`
```

Comment:
Not sure why this change, or why not have `#ifdef GC_SMALL_PAGE` here also?

Reply:
All files transitively included from `julia_threads.h` must be marked as `PUBLIC_HEADERS`.

If we added an `#ifdef GC_SMALL_PAGE` then we would need to include `src/options.h` (where this flag is defined) and mark it as a public header as well, which is a bit ugly.